libsolidity/lsp/Transport.cpp

```diff
@@ -41,6 +42,12 @@ using namespace std;
 using namespace solidity::lsp;
 
 // {{{ Transport
+ofstream Transport::traceLogFileStream() const
+{
+	lspAssert(m_traceLogFilePath, ErrorCode::InternalError, "");
```

Comment:
```suggestion
	lspAssert(m_traceLogFilePath.has_value(), ErrorCode::InternalError, "");
```

Actually, I wanted to suggest `.empty()` at first but then noticed it's an `optional`. Don't you also need a check against an empty path?

Reply:
Also, if the client sets the file path to empty, this will fail. I don't think you can assert here (or even have a validation against this). You just have to handle the case of not having a file name set.